src/libraries/Pausable.sol

```diff
@@ -26,17 +41,37 @@ abstract contract Pausable is Initializable {
      * @dev Contract state is paused, blocking
      */
     error RequireContractIsPaused();
-    // @dev Storage slot with the pause flag, this is managed by the `Pause` struct
 
-    struct PauseStorage {
-        uint256 _pauseFlag;
-    }
+    /**
+     * @dev the pauser is not a valid pauser account (e.g. `address(0)`)
+     */
+    error InvalidPauser(address account);
+
+    /**
+     * @dev Cannot renounce the pauser capability when the contract is in the `PAUSED` state
+     */
+    error CannotRenounceWhilePaused(address account);
 
     // @dev Emitted when the contract is paused
     event Paused(bool paused);
     event NotPaused(bool notPaused);
 
     bytes32 public constant PAUSE_SLOT = bytes32(uint256(keccak256("Pause.pauseFlag")) - 1);
+    bytes32 public constant PAUSER_ROLE_SLOT = bytes32(uint256(keccak256("Pause.pauseRole")) - 1);
+
+    function _getPauserStorage() private pure returns (PauserStorage storage $) {
+        uint256 slot = uint256(PAUSER_ROLE_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
+
+    /**
+     * @dev Returns the current pauser account address.
+     */
+    function _pauser() private view returns (address) {
```

Comment:
Worth having this method public as `pauser()` (without the leading underscore)? I feel like being able to read from elsewhere could be useful